elementary/monitor/data_monitoring.py

```diff
@@ -235,7 +235,11 @@ def generate_report(
             elementary_output_json_file.write(dumped_output_data)
 
         if should_open_browser:
-            webbrowser.open_new_tab("file://" + html_path)
+            try:
+                webbrowser.open_new_tab("file://" + html_path)
+            except webbrowser.Error as e:
+                logger.exception(f"Could not open {html_path} in the browser - Error: {e}")
```

Comment:
I think we can skip logging the exception itself using `logger.exception` as it produces very large, and a bit too detailed, tracebacks which are not interesting in this case.
This error can occur only on rare occasions which are mostly when you use WSL.

What do you think?

```suggestion
                logger.error("Unable to open the web browser.")
```

Reply:
Agree